### ISSUE TITLE
Image not appearing in Wordpress

### ISSUE THREAD

**Issue by romatou** [sachinchoolur/lightGallery#1201]
## Description

When I'm logged into Wordpress everything is fine, but if I logout the image is not opening in lightbox. The lightbox appears with black background. In code inspector there is a srcset next to src with a link ([screenshot attached](https://cln.sh/BSYv6D)).

### Steps to reproduce

Add lightgallery plugins to a Wordpress folder and add images through Gutenberg/Block Editor

### JS code that you use to initialize lightGallery.

```js
lightGallery(document.getElementById('content'), {
	thumbnail: true,
	selector: 'img',
});
```

### Sample HTML markup

```html
<main id="content">
    ...
        <figure class="wp-block-gallery">
            <ul>
                <li><figure><img src="img1.jpg"><figure></li>
                <li><figure><img src="img2.jpg"><figure></li>
                <li><figure><img src="img3.jpg"><figure></li>
            </ul>
        </figure>
    ...
</main>
```

### Environment

-   Browser and version - Safari 15.1
-   OS - macOS Monterey 12.0.1
-   lightGallery version - download as of November 5, 2021

**Comment by sachinchoolur** [comment]
Hey @romatou,

Could you please share the HTML generated by WP? You might have to set the selector differently

**Comment by romatou** [comment]
``` html

<div id="content">
	<figure class="wp-block-gallery columns-3 is-cropped">
		<ul class="blocks-gallery-grid">
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09910-682x1024.jpg" alt="" data-id="13097" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09910-scaled.jpg" data-link="https://site.local/?attachment_id=13097" class="wp-image-13097 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09910-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09910-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09910-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09910-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09910-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09910-scaled.jpg 1706w" data-lg-id="13e65561-9216-4706-8c04-e138cae1fdde" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09910-682x1024.jpg" alt="" data-id="13097" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09910-scaled.jpg" data-link="https://site.local/?attachment_id=13097" class="wp-image-13097" srcset="https://site.local/wp-content/uploads/2021/07/DSC09910-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09910-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09910-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09910-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09910-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09910-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript></figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09851-682x1024.jpg" alt="" data-id="13096" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09851-scaled.jpg" data-link="https://site.local/?attachment_id=13096" class="wp-image-13096 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09851-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09851-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09851-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09851-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09851-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09851-scaled.jpg 1706w" data-lg-id="ed797d97-bf94-4a9d-950e-f22c2b56c34d" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09851-682x1024.jpg" alt="" data-id="13096" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09851-scaled.jpg" data-link="https://site.local/?attachment_id=13096" class="wp-image-13096" srcset="https://site.local/wp-content/uploads/2021/07/DSC09851-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09851-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09851-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09851-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09851-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09851-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09792-682x1024.jpg" alt="" data-id="13095" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09792-scaled.jpg" data-link="https://site.local/?attachment_id=13095" class="wp-image-13095 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09792-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09792-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09792-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09792-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09792-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09792-scaled.jpg 1706w" data-lg-id="73ac28d1-af6d-467b-983e-63c81a313092" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09792-682x1024.jpg" alt="" data-id="13095" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09792-scaled.jpg" data-link="https://site.local/?attachment_id=13095" class="wp-image-13095" srcset="https://site.local/wp-content/uploads/2021/07/DSC09792-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09792-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09792-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09792-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09792-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09792-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09723-682x1024.jpg" alt="" data-id="13094" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09723-scaled.jpg" data-link="https://site.local/?attachment_id=13094" class="wp-image-13094 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09723-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09723-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09723-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09723-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09723-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09723-scaled.jpg 1706w" data-lg-id="2668e72b-5777-4e29-a2f8-71b6fa0ad90c" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09723-682x1024.jpg" alt="" data-id="13094" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09723-scaled.jpg" data-link="https://site.local/?attachment_id=13094" class="wp-image-13094" srcset="https://site.local/wp-content/uploads/2021/07/DSC09723-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09723-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09723-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09723-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09723-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09723-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09692-682x1024.jpg" alt="" data-id="13093" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09692-scaled.jpg" data-link="https://site.local/?attachment_id=13093" class="wp-image-13093 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09692-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09692-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09692-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09692-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09692-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09692-scaled.jpg 1706w" data-lg-id="0bc295f9-a225-4111-9f61-1e64fd215fa6" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09692-682x1024.jpg" alt="" data-id="13093" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09692-scaled.jpg" data-link="https://site.local/?attachment_id=13093" class="wp-image-13093" srcset="https://site.local/wp-content/uploads/2021/07/DSC09692-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09692-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09692-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09692-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09692-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09692-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09666-682x1024.jpg" alt="" data-id="13092" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09666-scaled.jpg" data-link="https://site.local/?attachment_id=13092" class="wp-image-13092 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09666-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09666-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09666-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09666-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09666-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09666-scaled.jpg 1706w" data-lg-id="59d027f6-a4a1-45de-97a4-abfff3169107" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09666-682x1024.jpg" alt="" data-id="13092" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09666-scaled.jpg" data-link="https://site.local/?attachment_id=13092" class="wp-image-13092" srcset="https://site.local/wp-content/uploads/2021/07/DSC09666-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09666-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09666-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09666-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09666-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09666-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript></figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="683" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09662-683x1024.jpg" alt="" data-id="13091" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09662.jpg" data-link="https://site.local/?attachment_id=13091" class="wp-image-13091 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09662-683x1024.jpg 683w, https://site.local/wp-content/uploads/2021/07/DSC09662-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09662-768x1152.jpg 768w" data-lg-id="9457d506-35f7-4e43-b595-8a81fabe0ebc" data-lazy-loaded="1" sizes="(max-width: 683px) 100vw, 683px"><noscript>&lt;img loading="lazy" width="683" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09662-683x1024.jpg" alt="" data-id="13091" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09662.jpg" data-link="https://site.local/?attachment_id=13091" class="wp-image-13091" srcset="https://site.local/wp-content/uploads/2021/07/DSC09662-683x1024.jpg 683w, https://site.local/wp-content/uploads/2021/07/DSC09662-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09662-768x1152.jpg 768w" sizes="(max-width: 683px) 100vw, 683px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure><img loading="eager" width="683" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09650-683x1024.jpg" alt="" data-id="13090" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09650-scaled.jpg" data-link="https://site.local/?attachment_id=13090" class="wp-image-13090 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09650-683x1024.jpg 683w, https://site.local/wp-content/uploads/2021/07/DSC09650-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09650-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09650-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09650-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09650-scaled.jpg 1707w" data-lg-id="f4927ad6-cae8-4c20-a628-09055b5de7b9" data-lazy-loaded="1" sizes="(max-width: 683px) 100vw, 683px"><noscript>&lt;img loading="lazy" width="683" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09650-683x1024.jpg" alt="" data-id="13090" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09650-scaled.jpg" data-link="https://site.local/?attachment_id=13090" class="wp-image-13090" srcset="https://site.local/wp-content/uploads/2021/07/DSC09650-683x1024.jpg 683w, https://site.local/wp-content/uploads/2021/07/DSC09650-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09650-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09650-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09650-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09650-scaled.jpg 1707w" sizes="(max-width: 683px) 100vw, 683px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="eager" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09610-682x1024.jpg" alt="" data-id="13089" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09610-scaled.jpg" data-link="https://site.local/?attachment_id=13089" class="wp-image-13089 jetpack-lazy-image jetpack-lazy-image--handled" srcset="https://site.local/wp-content/uploads/2021/07/DSC09610-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09610-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09610-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09610-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09610-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09610-scaled.jpg 1706w" data-lg-id="7cdb235e-ac23-4952-92d8-b8048fefe58a" data-lazy-loaded="1" sizes="(max-width: 682px) 100vw, 682px"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC09610-682x1024.jpg" alt="" data-id="13089" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC09610-scaled.jpg" data-link="https://site.local/?attachment_id=13089" class="wp-image-13089" srcset="https://site.local/wp-content/uploads/2021/07/DSC09610-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC09610-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC09610-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC09610-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC09610-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC09610-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
			<li class="blocks-gallery-item">
				<figure>
					<img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC00227-682x1024.jpg" alt="" data-id="13088" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC00227-scaled.jpg" data-link="https://site.local/?attachment_id=13088" class="wp-image-13088 jetpack-lazy-image" data-lazy-srcset="https://site.local/wp-content/uploads/2021/07/DSC00227-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC00227-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC00227-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC00227-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC00227-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC00227-scaled.jpg 1706w" data-lazy-sizes="(max-width: 682px) 100vw, 682px" data-lazy-src="https://site.local/wp-content/uploads/2021/07/DSC00227-682x1024.jpg?is-pending-load=1" srcset="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-lg-id="b9f4d11f-5a57-4b01-bd65-b85747fa95c3"><noscript>&lt;img loading="lazy" width="682" height="1024" src="https://site.local/wp-content/uploads/2021/07/DSC00227-682x1024.jpg" alt="" data-id="13088" data-full-url="https://site.local/wp-content/uploads/2021/07/DSC00227-scaled.jpg" data-link="https://site.local/?attachment_id=13088" class="wp-image-13088" srcset="https://site.local/wp-content/uploads/2021/07/DSC00227-682x1024.jpg 682w, https://site.local/wp-content/uploads/2021/07/DSC00227-200x300.jpg 200w, https://site.local/wp-content/uploads/2021/07/DSC00227-768x1152.jpg 768w, https://site.local/wp-content/uploads/2021/07/DSC00227-1024x1536.jpg 1024w, https://site.local/wp-content/uploads/2021/07/DSC00227-1365x2048.jpg 1365w, https://site.local/wp-content/uploads/2021/07/DSC00227-scaled.jpg 1706w" sizes="(max-width: 682px) 100vw, 682px" /&gt;</noscript>
				</figure>
			</li>
		</ul>
	</figure>
</div>

```

**Comment by suraj235** [comment]
Hi @sachinchoolur 
I am also experiencing a similar issue on my WordPress site. I had raised that [here](https://github.com/sachinchoolur/lightGallery/issues/1202)

Any updates?

**Comment by sachinchoolur** [comment]
Hey @romatou,

using the 'img' as selector might not be sufficient here as it does not have a `data-src` attribute. We might need an option to configure how to detect the image source. 

Is it possible to share the website URL?
Or at least share the generated HTML markup from both logged in and logged out states?

**Comment by romatou** [comment]
Here is a saved html markup from a WordPress page where I experience the problem: [index.txt](https://github.com/sachinchoolur/lightGallery/files/7513362/index.txt)

**Comment by romatou** [comment]
data:image/gif;base64 in srcset is the only difference between logged state and not

**Comment by sachinchoolur** [comment]
Can you try setting the anchor tag as selector? 

```js
lightGallery(document.getElementById('content'), {
	thumbnail: true,
	selector: 'a',
});
```

**Comment by tybears** [comment]
I was experiencing the same issue while using lightGallery in WordPress. I was seeing the same thing in developer tools (data:image/gif;base64).

 The issue for me was due to lazy loading. I turned off lazy loading and everything worked. However, I do need image lazy loading so I tried a few things to get it to work with it enabled.

I tried using a selector but still had the same issue. What worked was using exThumbImage.

I added this to my settings...
exThumbImage: 'data-thumb',

And this is the HTML...
<a href="/lightgallery/img/test.jpg" data-thumb="/lightgallery/img/test.jpg" ><img src="/lightgallery/img/test.jpg"></a>

If there is a better way to write the HTML portion where I don't have to write out the image 3 times I would appreciate any input.
